src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    static final int RESIZE_FACTOR = 2;
+    private Node<K,V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resize();
+        }
+        putToTable(key, value);
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+        if (currentNode == null) {
+            return null;
+        }
+        do {
+            if (currentNode.key == key || currentNode.key != null && currentNode.key.equals(key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        } while (currentNode != null);
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K,V>[] oldTable = table;
+        table = new Node[table.length * RESIZE_FACTOR];
+        size = 0;
+        for (int i = 0; i < oldTable.length; i++) {
+            if (oldTable[i] != null) {
+                Node<K, V> curentNode = oldTable[i];
+                Node<K, V> curentNodeNext;
+                do {
+                    curentNodeNext = curentNode.next;
+                    putToTable(curentNode.key, curentNode.value);
+                    curentNode = curentNodeNext;
+
+                } while (curentNode != null);
+            }
+
+        }
+        threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+    }
```

Comment:
```suggestion
    private void resize() {
        Node<K, V>[] oldTable = table;
        table = new Node[table.length * RESIZE_FACTOR];
        size = 0;
        for (Node<K, V> headNode : oldTable) {
            while (headNode != null) {
                putToTable(headNode.key, headNode.value);
                headNode = headNode.next;
            }
        }
    }
```
if you wish you can reuse putToTable here